From: Moshe Lazer <moshel@mellanox.com>
Subject: [PATCH] BACKPORTS: mlx4_core add pci_physfn to earlier Linux kernels

pci_physfn function introduced in kernel 2.6.35
This patch provides this function to earlier Linux kernels

issue: 120048

Signed-off-by: Moshe Lazer <moshel@mellanox.com>
---
 drivers/net/ethernet/mellanox/mlx4/main.c |   15 +++++++++++++++
 1 files changed, 15 insertions(+), 0 deletions(-)

diff --git a/drivers/net/ethernet/mellanox/mlx4/main.c b/drivers/net/ethernet/mellanox/mlx4/main.c
index xxxxxxx..xxxxxxx xxxxxx
--- a/drivers/net/ethernet/mellanox/mlx4/main.c
+++ b/drivers/net/ethernet/mellanox/mlx4/main.c
@@ -536,6 +536,21 @@ static void process_mod_param_profile(struct mlx4_profile *profile)
 	}
 }
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,35)
+#ifndef CONFIG_COMPAT_IS_PCI_PHYSFN
+/* This function copied from linux/pci.h (kernel 2.6.35) */
+static inline struct pci_dev *pci_physfn(struct pci_dev *dev)
+{
+#ifdef CONFIG_PCI_IOV
+	if (dev->is_virtfn)
+		dev = dev->physfn;
+#endif
+
+	return dev;
+}
+#endif
+#endif
+
 int mlx4_check_port_params(struct mlx4_dev *dev,
 			   enum mlx4_port_type *port_type)
 {
